web/ui/react-app/src/pages/flags/Flags.tsx

```diff
@@ -65,6 +61,10 @@ export const FlagsContent: FC<FlagsProps> = ({ data = {} }) => {
     focused: true,
   };
   const [sortState, setSortState] = useState(initialSort);
+  const searchable = Object.entries(data)
+    .sort(compareAlphaFn(sortState.name === 'Flag', !sortState.alpha))
```

Comment:
I noticed that the user-chosen sorting is introducing a new awkwardness when fuzzy-searching, where you expect the fuzzy-matching quality to determine sort order. Otherwise when you want "retention" and type "reten", you get:

![reten](https://user-images.githubusercontent.com/538008/111912903-f6b31f80-8a6b-11eb-993a-13d5f8ee6e1c.png)

I think it would be best to turn off the user-chosen sorting when there is anything in the search input field and use the fuzzy matching score instead (like in the old /graph UI, see https://github.com/prometheus/prometheus/blob/2ddecf92a235f71b7757e545afd80a233d1a1c99/web/ui/static/js/graph/index.js#L317-L320).

Btw. I noticed that in the new React UI in the PromQL input field, this sorting is also not done for metrics autocompletion - it doesn't use the fuzzy score for sorting there, but just the original order. But that will fix itself by me switching the whole expression input over to https://github.com/prometheus-community/codemirror-promql in the next weeks. The CodeMirror plugin also uses a better Fuzzy lib fork by @Nexucis: https://github.com/Nexucis/fuzzy - that one has better matching/scoring and is more maintained now :)

Reply:
thank you for the advertising @juliusv :).

Just to know, would like to replace the Fuzzy lib with the other one ? And if yes do you expect it would be done in this PR, or in another one ?

Well because if it is expecting to be done in another one, maybe I could try to do it.

Reply:
the lib has been replaced and when the list is filtered it is using the sorting from the `fuzzy` lib. Otherwise it is using the sorting from the column